### DISCUSSION TITLE
Using an actual Button component to trigger Disclosure

### DISCUSSION THREAD

**Discussion by giantrobotbee** [ariakit/ariakit#3231] [Questions] [answered]
I have a UI where a Button™ click should trigger the show/hide of a Disclosure (which contains some form fields). I have a separate Button component I'd like to use for this. It appears that `Disclosure` creates its own button for `Disclosure` which wraps my button when used like this

```    <DisclosureProvider>
      <AriaDisclosure className={triggerClasses}>
        <MyButton />
      </AriaDisclosure>
      <DisclosureContent className={contentClasses}>
        <p>Hellote</p>
      </DisclosureContent>
    </DisclosureProvider>
```

I see the `triggerOnClick` prop, but no good examples of using it. Would I need to eschew the provider component and use the state hook instead? Or am I missing something?

**Comment by diegohaz** [comment, marked as answer]
You can use the [`render`](https://ariakit.org/guide/composition) prop:

```jsx
<Ariakit.Disclosure render={<MyButton />}>
  Open
</Ariakit.Disclosure>
```

Make sure `MyButton` is [open for extension](https://ariakit.org/guide/composition#custom-components-must-be-open-for-extension).

**Reply by giantrobotbee** [reply]
TY!
